### DISCUSSION TITLE
Bitfield field selection syntax

### DISCUSSION THREAD

**Discussion by Timmmm** [rems-project/sail#774] [General]
Sail currently uses `mstatus[MIE]` syntax for bitfields (formerly `mstatus->MIE()` IIRC). But in my experience most people writing pseudocode use `mstatus.MIE`, and it is also more consistent with most programming languages (e.g. C).

Probably shouldn't change the syntax it a second time! I was just wondering what the motivation was. I guess it's consistent with `mstatus[4]` or whatever?

**Comment by Alasdair** [comment]
Yes, keeping it the same as the bitslice syntax was the idea.

I'm not against changing it again if the alternative would be more intuitive. Writing a tool to convert automatically wouldn't be too hard.
